x-pack/legacy/plugins/lens/public/editor_frame_plugin/plugin.tsx

```diff
@@ -128,3 +142,60 @@ export const editorFrameStop = () => editorFrame.stop();
 function NotFound() {
   return <h1>TODO: 404 Page</h1>;
 }
+
+function is404(error?: unknown) {
+  return error && (error as { statusCode: number }).statusCode === 404;
+}
+
+export async function init({
+  persistedId,
+  store,
+  onError,
+}: InitializationProps): Promise<InitializationResult> {
+  if (!persistedId) {
+    return {};
+  } else {
+    return store
+      .load(persistedId)
+      .then(doc => ({ doc }))
+      .catch((error: Error) => {
+        if (!is404(error)) {
+          onError(error);
+        }
+        return { error };
+      });
```

Comment:
This is one of those cases where I find the promise chain to be much more readable. We could remove the async signature, but it would make line 156 a bit messier.